src/index.js

```diff
@@ -17,65 +18,38 @@ const DEFAULT_OPTIONS = {
 };
 
 module.exports = function pluginGraphQLDocGenerator(context, opts) {
-  // Merge defaults with user-defined options.
-  const config = { ...DEFAULT_OPTIONS, ...opts };
+  // Merge defaults with user-defined options in config file.
+  pluginConfigs.push({ ...DEFAULT_OPTIONS, ...opts });
```

Comment:
`pluginConfigs`should be immutable. This can be done by using a builder method. 
At the same time, we can also move `DEFAULT_OPTIONS` and the merge logic with it to `config.js`.

```js
const path = require("path");
const os = require("os");

const DEFAULT_OPTIONS = {
  schema: "./schema.graphl",
  rootPath: "./docs",
  baseURL: "schema",
  linkRoot: "/",
  homepage: path.join(__dirname, "../assets/", "generated.md"),
  diffMethod: "SCHEMA-DIFF",
  tmpDir: path.join(os.tmpdir(), "@edno/docusaurus2-graphql-doc-generator"),
  loaders: {},
  pretty: false,
};

const pluginConfigs = [];

function addConfig(options) {
  pluginConfigs.push({ ...DEFAULT_OPTIONS, ...options });
  return pluginConfigs;
}
```

Reply:
Cool. Addressed in 9b550e2 (there is no array anymore).